ballerina/dispatcher_service.bal

```diff
@@ -0,0 +1,324 @@
+            self.mcpService = mcpService;
+        }
+    }
+
+    isolated function removeServiceRef() {
+        lock {
+            self.mcpService = ();
+        }
+    }
+
+    isolated function setServerConfigs(ServerConfiguration serverConfigs) {
+        lock {
+            self.serverConfigs = serverConfigs.cloneReadOnly();
+        }
+    }
+
+    isolated resource function post .(@http:Payload JsonRpcMessage request, http:Headers headers)
+            returns http:BadRequest|http:NotAcceptable|http:UnsupportedMediaType|http:Accepted|http:Ok|ServerError {
+
+        http:NotAcceptable|http:UnsupportedMediaType? headerValidationError = self.validateHeaders(headers);
+        if !(headerValidationError is ()) {
+            return headerValidationError;
+        }
+
+        lock {
+            if request is JsonRpcRequest {
+                return self.processJsonRpcRequest(request.cloneReadOnly());
+            }
+            else if request is JsonRpcNotification {
+                return self.processJsonRpcNotification(request.cloneReadOnly());
+            }
+
+            JsonRpcError jsonRpcError = self.createErrorResponse(null, INVALID_REQUEST, "Unsupported request type");
+            return <http:BadRequest>{
+                body: jsonRpcError.cloneReadOnly()
+            };
+        }
+    }
+
+    private isolated function validateHeaders(http:Headers headers)
+            returns http:NotAcceptable|http:UnsupportedMediaType? {
+
+        // Validate Accept header
+        string|http:HeaderNotFoundError acceptHeader = headers.getHeader(ACCEPT_HEADER);
+        if acceptHeader is http:HeaderNotFoundError {
+            JsonRpcError jsonRpcError = self.createErrorResponse(1, NOT_ACCEPTABLE,
+                "Not Acceptable: Client must accept both application/json and text/event-stream");
+            return <http:NotAcceptable>{
+                body: jsonRpcError
+            };
+        }
+
+        if !acceptHeader.includes(CONTENT_TYPE_JSON) || !acceptHeader.includes(CONTENT_TYPE_SSE) {
+            JsonRpcError jsonRpcError = self.createErrorResponse(1, NOT_ACCEPTABLE,
+                "Not Acceptable: Client must accept both application/json and text/event-stream");
+            return <http:NotAcceptable>{
+                body: jsonRpcError
+            };
+        }
+
+        // Validate Content-Type header
+        string|http:HeaderNotFoundError contentTypeHeader = headers.getHeader(CONTENT_TYPE_HEADER);
+        if contentTypeHeader is http:HeaderNotFoundError {
+            JsonRpcError jsonRpcError = self.createErrorResponse(null, UNSUPPORTED_MEDIA_TYPE,
+                "Unsupported Media Type: Content-Type must be application/json");
+            return <http:UnsupportedMediaType>{
+                body: jsonRpcError
+            };
+        }
+
+        if !contentTypeHeader.includes(CONTENT_TYPE_JSON) {
+            JsonRpcError jsonRpcError = self.createErrorResponse(null, UNSUPPORTED_MEDIA_TYPE,
```

Comment:
```suggestion
            JsonRpcError jsonRpcError = self.createErrorResponse((), UNSUPPORTED_MEDIA_TYPE,
```
in Ballerina, we prefer using `()` to represent the absence of a value, and reserve `null` only for cases involving raw json handling.